guidelines/enhancement_template.md

```diff
@@ -115,6 +117,17 @@ bogged down.
 
 Include a story on how this proposal will be operationalized:  lifecycled, monitored and remediated at scale.
 
+### API Extensions
+
+- name the API extensions (CRDs, webhooks, aggregated API servers) this enhancement 
+- adds or modifies
+- does this enhancement modify the behaviour of existing resources and if yes how
+
+  Examples:
+    - adds a finalizer to namespaces. Namespace cannot be deleted without our controller running.
+    - restricts the label format for objects to X.
+    - defaults field Y on object kind Z.
```

Comment:
would you also expect new resource types, and/or new fields being added to existing resource types, to be described here?

(I would, though they also tend to end up in the implementation details section currently)

Reply:
hm, I don't see why "who owns the api" vs "who is writing the EP" matters?  That may affect who needs to review it, but not what the design needs to cover.

what matters is the kind of change you are proposing to the product, and the effects it may have that need to be considered.

For which it seems like "api" changes need to be broken down into a few categories, e.g.:

1) new api type
2) updating existing api type (e.g. new fields)
3) adding a webhook (not sure if conversion vs admission need to be divided)
4) adding/changing validations on an apitype (adding is usually not allowed, of course)
5) adding/changing rbac associated w/ an api type

i'm not saying you need to cover all of them in your changes here, but i think it's important we make it clear which kinds of changes these new sections are applicable to.